controllers/progressiverollout_controller_unit_test.go

```diff
@@ -0,0 +1,56 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	applicationpkg "github.com/argoproj/argo-cd/pkg/apiclient/application"
+	argov1alpha1 "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	. "github.com/onsi/gomega"
+	"google.golang.org/grpc"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+type mockArgoCDAppClientSyncOK struct{}
```

Comment:
I think there is something that I'm missing here: why we don't have a `mockArgoCDAppClient` package with its own methods and tests?
The approach would be similar in https://github.com/Skyscanner/kms-issuer/tree/main/pkg/kmsmock

Reply:
For why the mocks are in the file and not a separate package, see the other comment - I'm happy to move them if we don't want to.

There's also the fact that I want different mocks to do different things - in fact, the one in `suite_test` is more like a stub than a mock _per se_, it's there just to make sure the other tests pass by default.